backends/core/workers.py

```diff
@@ -758,3 +761,76 @@ def _execute(self):
     self.retry(request.execute)()
     job_name = '%s/jobs/%s' % (project_id, self._ml_job_id)
     self._enqueue('MLWaiter', {'job_name': job_name}, 60)
+
+
+class MeasurementProtocolWorker(Worker):
+  """Abstract Measurement Protocol worker."""
+
+  def _send_event_hit(self, user_agent='CRMint / 0.1', **kwargs):
+    """Send a measurement protocol hit.
+
+    Arguments:
+      user_agent string representing the client User Agent.
+      kwargs dictionary of key/values to pass to the Measurement
+          Protocol endpoint.
+
+    Raises: WorkerException if the HTTP request fails.
+    """
+    headers = {'user-agent': user_agent}
+    payload = {'v': 1}
+    payload.update(kwargs)
+
+    req = requests.post('https://www.google-analytics.com/collect',
+                        headers=headers,
+                        data=payload)
+
+    if req.status_code != requests.codes.ok:
+      raise WorkerException('Failed to send event hit with status code (%s) '
```

Comment:
I've used the `retry` method. The only caveat that I see is that if one http request fails repeatedly then the whole job will be jeopardize since it will increase the total time by a lot.

We could also just silently fail by just logging the error. What do you think?